src/main/java/org/opentripplanner/updater/spi/TimetableSnapshotFlush.java

```diff
@@ -0,0 +1,43 @@
+package org.opentripplanner.updater.spi;
+
+import org.opentripplanner.ext.siri.SiriTimetableSnapshotSource;
+import org.opentripplanner.updater.trip.TimetableSnapshotSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Flush the timetable snapshot buffer by committing pending changes.
+ * Exceptions occurring during the flush are caught and ignored: the scheduler can then retry
+ * the task later.
+ */
+public class TimetableSnapshotFlush implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TimetableSnapshotFlush.class);
+
+  private final SiriTimetableSnapshotSource siriTimetableSnapshotSource;
+  private final TimetableSnapshotSource gtfsTimetableSnapshotSource;
+
+  public TimetableSnapshotFlush(
+    SiriTimetableSnapshotSource siriTimetableSnapshotSource,
+    TimetableSnapshotSource gtfsTimetableSnapshotSource
+  ) {
+    this.siriTimetableSnapshotSource = siriTimetableSnapshotSource;
+    this.gtfsTimetableSnapshotSource = gtfsTimetableSnapshotSource;
+  }
+
+  @Override
+  public void run() {
+    try {
+      LOG.debug("Flushing timetable snapshot buffer");
+      if (siriTimetableSnapshotSource != null) {
+        siriTimetableSnapshotSource.flushBuffer();
+      }
+      if (gtfsTimetableSnapshotSource != null) {
+        gtfsTimetableSnapshotSource.flushBuffer();
+      }
+      LOG.debug("Flushed timetable snapshot buffer");
+    } catch (Throwable t) {
+      LOG.error("Error flushing timetable snapshot buffer", t);
```

Comment:
The error message is missing the placeholder for the Throwable.toString result.
```suggestion
      LOG.error("Error flushing timetable snapshot buffer: {}", t);
```

As a suggestion (not necessary for approval of this particular PR): Given that these log messages should be rare and triggered at most once every few seconds, they shouldn't generate excessive log noise, so it might make sense to log a full stack trace instead of the short exception summary to facilitate debugging. A utility method to do so could be reused anywhere we have background tasks running in an executor. Example at: https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/util/ExceptionUtils.java#L23

Reply:
The logger already logs the full stacktrace when the last vararg is a Throwable.
The log message looks like this:
```
09:54:25.602 ERROR [graph-writer]  (TimetableSnapshotFlush.java:43) Error flushing timetable snapshot buffer
java.lang.RuntimeException: EXCEPTION WHILE FLUSHING
	at org.opentripplanner.updater.spi.TimetableSnapshotFlush.run(TimetableSnapshotFlush.java:33)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:358)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

Reply:
Ah my mistake, looks like auto-logging exceptions as stack traces has been a feature for a long time. I shouldn't have assumed the placeholder was necessary.